pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java

```diff
@@ -72,11 +74,30 @@ public SegmentProcessorFramework(List<RecordReader> recordReaders, SegmentProces
       File workingDir)
       throws IOException {
     Preconditions.checkState(!recordReaders.isEmpty(), "No record reader is provided");
-
     LOGGER.info("Initializing SegmentProcessorFramework with {} record readers, config: {}, working dir: {}",
         recordReaders.size(), segmentProcessorConfig, workingDir.getAbsolutePath());
 
     _recordReaders = recordReaders;
+    _recordReaderFileConfigs = null;
+    _segmentProcessorConfig = segmentProcessorConfig;
+
+    _mapperOutputDir = new File(workingDir, "mapper_output");
+    FileUtils.forceMkdir(_mapperOutputDir);
+    _reducerOutputDir = new File(workingDir, "reducer_output");
+    FileUtils.forceMkdir(_reducerOutputDir);
+    _segmentsOutputDir = new File(workingDir, "segments_output");
+    FileUtils.forceMkdir(_segmentsOutputDir);
+  }
+
+  public SegmentProcessorFramework(SegmentProcessorConfig segmentProcessorConfig, File workingDir,
```

Comment:
Yes thats what I did initially :), but java's type erasure shows that the 2 constructors clash, since the have the same arguments in the same order (1st one being List<?>). And I would really like to have 2 constructors for this, to make the usage explicit.